### PR TITLE
fix: [M3-7356] - VPC IPv4 checkboxes in Linode Config dialog

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Fix VPC IPv4 checkbox and input display when editing a config that already has ipv4 data

## Changes  🔄
1. The "Assign a public IPv4 address" checkbox is now checked in the Edit Config dialog upon opening if a VPC interface has already been 1:1 NAT’ed (i.e., if the `ipv4.nat_1_1` field is populated)
2. The "Auto-assign a VPC IPv4 address" is now unchecked in the Edit Config dialog if the `ipv4.vpc` field is already populated and we just expose the VPC IPv4 input field (pre-filled with the `ipv4.vpc` value)

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115299789/39dbe0cc-5147-4b55-b499-b0f1ac55d205" /> | <video src="https://github.com/linode/manager/assets/115299789/b1e708d5-5628-4e1c-b35c-461f535c9070" /> |

(Linode was deleted after)

### Prerequisites
- Point to the dev env and ensure your account has vpc customer tags
- Create a VPC and Subnet if you don't already have one
- Click on Add Configuration in the Linode details -> Configuration tab
  - Provide a label and scroll down to the Networking section
  - Add a VPC, Subnet, and check both of the checkboxes

### Reproduction steps
- Be on the develop branch or go to the remote dev environment
- Filter your Network requests by `config` and notice how the `ipv4.vpc` and `ipv4.nat_1_1` fields of the interface are populated
- Click on Edit Configuration. The "Auto-assign a VPC IPv4 address" checkbox is checked and the VPC IPv4 input field is not exposed despite the field being populated. And the "Assign a public IPv4 address" checkbox is unchecked

### Verification steps 
- Filter your Network requests by `config` and notice how the `ipv4.vpc` and `ipv4.nat_1_1` fields of the interface are populated
- Click on Edit Configuration. The "Auto-assign a VPC IPv4 address" checkbox should be unchecked and the VPC IPv4 input field is now exposed and populated with the `ipv4.vpc` value. And the "Assign a public IPv4 address" checkbox is now checked
 ```
yarn test VPCPanel
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeSettings/InterfaceSelect.tsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ interface VPCStateErrors {
 
 interface VPCState {
   errors: VPCStateErrors;
+  iPv4_nat_1_1?: string;
```
Comment (hana-linode) [inline packages/manager/src/features/Linodes/LinodesDetail/LinodeSettings/InterfaceSelect.tsx:43]: Open to better variable naming

File: packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.test.tsx [hunk 2]
```diff
@@ -128,4 +128,68 @@ describe('VPCPanel', () => {
...
+      queryClient,
+    });
+
+    await waitFor(() => {
+      // the "Auto-assign a VPC IPv4 address for this Linode in the VPC" checkbox is the first one (0 index)
+      expect(wrapper.getAllByRole('checkbox')[0]).not.toBeChecked();
+
+      wrapper.getByText('VPC IPv4');
+      const vpcIPv4Input = wrapper.getAllByRole('textbox')[2];
+      expect(vpcIPv4Input).toHaveValue('10.0.4.3');
```
Comment (jdamore-linode) [inline packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.test.tsx:163]: Selecting the checkbox and input using their label text might make the tests a little more readable and a little less fragile to future changes

```suggestion
      expect(wrapper.getByLabelText('Auto-assign a VPC IPv4 address for this Linode in the VPC')).not.toBeChecked();
      // Using regex here to account for the "(required)" indicator.
      expect(wrapper.getByLabelText(/^VPC IPv4.*/)).toHaveValue('10.0.4.3');
```

File: packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.test.tsx [hunk 3]
```diff
@@ -128,4 +128,68 @@ describe('VPCPanel', () => {
...
+      })
+    );
+
+    const wrapper = renderWithTheme(<VPCPanel {..._props} />, {
+      flags: { vpc: true },
+      queryClient,
+    });
+
+    await waitFor(() => {
+      expect(wrapper.getAllByRole('checkbox')[1]).toBeChecked();
```
Comment (jdamore-linode) [inline packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.test.tsx:192]: Same as above!

```suggestion
      expect(wrapper.getByLabelText('Assign a public IPv4 address for this Linode')).toBeChecked();
```

**Review by jdamore-linode — APPROVED** [review]
Thanks @hana-linode! Also double checked that all of the VPC and Linode config Cypress tests are continuing to pass. Posted a couple opinionated suggestions involving the unit test input/checkbox element selectors but they're super minor!

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeSettings/InterfaceSelect.tsx (hunk 1 above)
Comment (dwiley-akamai) [reply packages/manager/src/features/Linodes/LinodesDetail/LinodeSettings/InterfaceSelect.tsx:43]: `nattedIPv4Address` 👀

**Review by dwiley-akamai — APPROVED** [review]
"Auto-assign a VPC IPv4 address" checkbox is unchecked and the "VPC IPv4" input field is exposed and populated with `ipv4.vpc` value when that field has a value ✅

"Assign a public IPv4 address" checkbox is checked when `ipv4.nat_1_1` has a value ✅

Code review ✅

**Comment by tyler-akamai** [comment]
- [ ] ipv4.vpc and ipv4.nat_1_1 fields of the interface are populated
- [ ] The "Auto-assign a VPC IPv4 address" checkbox is unchecked and the VPC IPv4 input field is exposed and populated with the ipv4.vpc value. And the "Assign a public IPv4 address" checkbox is checked
- [ ] yarn test VPCPanel

**Pull request merged**